modules/security/src/main/java/io/oasp/module/security/common/api/accesscontrol/AccessControlGroup.java

```diff
@@ -141,20 +135,11 @@ public boolean equals(Object obj) {
     if (!super.equals(obj)) {
       return false;
     }
-    if (getClass() != obj.getClass()) {
-      return false;
-    }
     AccessControlGroup other = (AccessControlGroup) obj;
```

Comment:
Wrong. ```AccessControlGroup``` extends ```AccessControl``` that already contains the same check in ```equals```. As ```super.equals(obj)``` has to be true at this point the if check was redundant and pointless here.